frontend/src/app/data-entry/agent-form/agent-description-form/agent-description-form.component.ts

```diff
@@ -0,0 +1,112 @@
+    styleUrls: ['./agent-description-form.component.scss'],
+})
+export class AgentDescriptionFormComponent implements OnChanges, OnDestroy {
+    @Input() id?: string;
+
+    genderOptions: { value: GenderChoices, label: string }[] = [
+        { value: GenderChoices.Female, label: 'Female' },
+        { value: GenderChoices.Male, label: 'Male' },
+        { value: GenderChoices.Other, label: 'Other' },
+        { value: GenderChoices.Mixed, label: 'Mixed (for groups)' },
+        { value: GenderChoices.Unknown, label: 'Unknown' }
+    ];
+
+    genderSourceMentionOptions: { value: GenderSourceMentionChoices, label: string }[] = [
+        { value: GenderSourceMentionChoices.Direct, label: 'Mentioned' },
+        { value: GenderSourceMentionChoices.Implied, label: 'Implied' },
+    ];
+
+    locationSourceMentionOptions: { value: LocationSourceMentionChoices, label: string }[] = [
+        { value: LocationSourceMentionChoices.Direct, label: 'Mentioned' },
+        { value: LocationSourceMentionChoices.Implied, label: 'Implied' },
+    ];
+
+    form = new FormGroup({
+        designators: new FormControl<string[]>([], { nonNullable: true }),
+        gender: new FormGroup({
+            gender: new FormControl<string>(GenderChoices.Unknown),
+            sourceMention: new FormControl<string>(GenderSourceMentionChoices.Direct),
+            note: new FormControl<string>(''),
+        }),
+        location: new FormGroup({
+            hasLocation: new FormControl<boolean>(false, { nonNullable: true }),
+            location: new FormControl<string | null>(null),
+            sourceMention: new FormControl<string>(LocationSourceMentionChoices.Direct),
+            note: new FormControl<string>(''),
+        })
+    })
+
+    isGroup$: Observable<boolean>;
+    locations$: Observable<LocationsInSourceListQuery>;
+
+    private id$ = new Subject<string>();
+    private data$: Observable<DataEntryAgentDescriptionQuery>;
+
+    constructor(
+        private agentQuery: DataEntryAgentDescriptionGQL,
+        private locationsQuery: LocationsInSourceListGQL,
+    ) {
+        this.data$ = this.id$.pipe(
+            switchMap(id => this.agentQuery.watch({ id }).valueChanges),
+            map(result => result.data),
+            shareReplay(1),
+        );
+        this.isGroup$ = this.data$.pipe(
+            map(result => result.agentDescription?.isGroup || false),
+        );
+        this.locations$ = this.data$.pipe(
+            map(data => data.agentDescription?.source.id),
+            filter(_.negate(_.isUndefined)),
+            switchMap(id => this.locationsQuery.watch({ id }).valueChanges),
+            map(result => result.data),
+            shareReplay(1),
+        );
+        this.data$.subscribe(this.updateFormData.bind(this));
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes['id'] && this.id) {
+            this.id$.next(this.id);
+        }
+    }
```

Comment:
One more thing I just realised: instead of passing the ID around via inputs, you could have every sub-form just take it from the route again as you do it in the main form. That gets you rid of this `ngOnChanges` and the subject.